central/image/service/http_handler.go

```diff
@@ -0,0 +1,97 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/enrichment"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/images/enricher"
+	"github.com/stackrox/rox/pkg/images/integration"
+	"github.com/stackrox/rox/pkg/scanners/types"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+)
+
+type sbomHandler struct {
+	integration integration.Set
+	enricher    enricher.ImageEnricher
+}
+
+type SBOMRequestBody struct {
+	ClusterID string `json:"clusterID"`
+	ImageName string `json:"imageName"`
+	force     bool   `json:"force"`
+}
+
+func (s *sbomHandler) ImageHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+		var params SBOMRequestBody
+		err := json.NewDecoder(r.Body).Decode(&params)
+		if err != nil {
+			httputil.WriteGRPCStyleError(w, codes.Internal, err)
+			return
+		}
+		bytes, err := s.getSbom(params, r.Context())
+		if err != nil {
+			httputil.WriteGRPCStyleError(w, codes.Internal, err)
+			return
+		}
+
+		// Tell the browser this is a download.
+		w.Header().Add("Content-Disposition", fmt.Sprintf(`attachment; filename="values-%s.json"`, "sbom.json"))
+		w.Header().Add("Content-Type", "text/yaml")
+		w.Header().Add("Content-Length", fmt.Sprint(len(bytes)))
+		_, _ = w.Write(bytes)
+		return
+	}
+}
+
+func (s *sbomHandler) enrichImage(params SBOMRequestBody, ctx context.Context) (*storage.Image, error) {
+	enrichmentCtx := enricher.EnrichmentContext{
+		FetchOpt:  enricher.UseCachesIfPossible,
+		Delegable: true,
+	}
+
+	if params.force {
+		enrichmentCtx.FetchOpt = enricher.UseImageNamesRefetchCachedValues
+	}
+
+	if params.ClusterID != "" {
+		enrichmentCtx.ClusterID = params.ClusterID
```

Comment:
Change this to derive the cluster ID from the request, the request may be a cluster name or id.  Example of how to do this in the image service:

https://github.com/stackrox/stackrox/blob/5ae72703c6512e1672a928b02fd4eb7fcdffc756/central/image/service/service_impl.go#L392